apps/client/src/pages/community/community-edit/community-edit.tsx

```diff
@@ -30,20 +32,33 @@ const COMMUNITY_CONTENT = {
 
 const CommunityEdit = () => {
   const navigate = useNavigate();
-  const [isDisabled, setIsDisabled] = useState(true);
   const queryClient = useQueryClient();
   const { postId } = useParams<{ postId: string }>();
   const location = useLocation();
-  const state = location.state as { title: string; content: string };
+  const state = location.state as {
+    title: string;
+    content: string;
+    category?: { category: string; description: string };
+  };
   const [title, setTitle] = useState(state.title);
   const [content, setContent] = useState(state.content);
+  const [category, setCategory] = useState(() => {
+    if (state.category) {
+      return (
+        categoryOptions.find(
+          (option) => option.value === state.category?.category,
+        ) || null
+      );
+    }
+    return null;
+  });
```

Comment:
```suggestion
const getInitialCategory = () => {
  if (!state.category) return null;

  const matchedOption = categoryOptions.find(
    (option) => option.value === state.category.category
  );

  return matchedOption ?? null;
};

const [category, setCategory] = useState(getInitialCategory);
```

이렇게도 작성 가능할까요? 결국 우리가 알고싶은 값은 category 같은데 그 과정이 카테고리를 설명하는 코드라인에 포함되어있어서 선언적으로 읽히진 않는 것 같아요

Reply:
코드는 그냥 예시일뿐이고, 가능하다면 es-toolkit 을 도입해서, 복잡한 로직을 줄이는 것도 대찬성이에요

Reply:
good 입니다 요거 오늘 안으로 처리 해둘게요